haiku/_src/initializers_test.py

```diff
@@ -61,5 +62,45 @@ def test_initializers(self, dtype):
       self.assertEqual(generated.shape, shape)
       self.assertEqual(generated.dtype, dtype)
 
+  @test_utils.transform_and_run
+  def test_invalid_variance_scale(self):
+
+    with self.assertRaises(ValueError):
```

Comment:
Please test the error message is as expected here. For example:

```python
with self.assertRaisesRegex(ValueError, "scale.*must be a positive float"):
  initializers.VarianceScaling(scale=-1.0)
```

Same for the two below.